test/cppsuite/CMakeLists.txt

```diff
@@ -28,38 +28,57 @@
 
 project(cppsuite CXX)
 
-create_test_executable(run
-    SOURCES test_harness/core/component.cxx 
-            test_harness/core/configuration.cxx 
-            test_harness/core/throttle.cxx 
-            test_harness/util/logger.cxx
-            test_harness/util/scoped_types.cxx 
-            test_harness/workload/database_model.cxx 
-            test_harness/workload/database_operation.cxx 
-            test_harness/workload/random_generator.cxx 
-            test_harness/workload/thread_context.cxx 
-            test_harness/workload/workload_tracking.cxx 
-            test_harness/workload/workload_validation.cxx 
-            test_harness/checkpoint_manager.cxx 
-            test_harness/connection_manager.cxx 
-            test_harness/runtime_monitor.cxx 
-            test_harness/test.cxx 
-            test_harness/thread_manager.cxx 
-            test_harness/timestamp_manager.cxx 
-            test_harness/workload_generator.cxx 
-            tests/run.cxx
-    FLAGS -std=c++11
-    INCLUDES ${CMAKE_CURRENT_SOURCE_DIR}
+# Create an intermediate static lib.
+add_library(cppsuite_test_harness STATIC
+    ../utility/test_util.h
+    test_harness/core/component.cxx 
+    test_harness/core/configuration.cxx 
+    test_harness/core/throttle.cxx 
+    test_harness/util/logger.cxx
+    test_harness/util/scoped_types.cxx 
+    test_harness/workload/database_model.cxx 
+    test_harness/workload/database_operation.cxx 
+    test_harness/workload/random_generator.cxx 
+    test_harness/workload/thread_context.cxx 
+    test_harness/workload/workload_tracking.cxx 
+    test_harness/workload/workload_validation.cxx 
+    test_harness/checkpoint_manager.cxx 
+    test_harness/connection_manager.cxx 
+    test_harness/runtime_monitor.cxx 
+    test_harness/test.cxx 
+    test_harness/thread_manager.cxx 
+    test_harness/timestamp_manager.cxx 
+    test_harness/workload_generator.cxx 
+)
+
+target_include_directories(cppsuite_test_harness PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
+target_compile_options(cppsuite_test_harness PUBLIC -std=c++11)
+target_link_libraries(cppsuite_test_harness PRIVATE test_util)
+
+# Create the different executables.
+create_test_executable(run 
+    SOURCES tests/run.cxx
+    LIBS cppsuite_test_harness
     ADDITIONAL_DIRECTORIES ${CMAKE_CURRENT_SOURCE_DIR}/configs
     CXX
 )
 
```

Comment:
Can we get the comment from the makefile copy and pasted here? The "try not to do this" one.

Reply:
Sure, 900725553b6e8a6e49cab101e37e63f1b22df402